src/parser/mod.rs

```diff
@@ -10319,6 +10321,104 @@ impl<'a> Parser<'a> {
+            // https://storage.googleapis.com/gweb-research2023-media/pubtools/1004848.pdf
+            while self.consume_token(&Token::VerticalBarRightAngleBracket) {
+                let kw = self.expect_one_of_keywords(&[
+                    Keyword::SELECT,
+                    Keyword::EXTEND,
+                    Keyword::SET,
+                    Keyword::DROP,
+                    Keyword::AS,
+                    Keyword::WHERE,
+                    Keyword::LIMIT,
+                    Keyword::AGGREGATE,
+                    Keyword::ORDER,
+                ])?;
+                match kw {
+                    // SELECT <expr> [[AS] alias], ...
+                    Keyword::SELECT => {
+                        let exprs = self.parse_comma_separated(Parser::parse_select_item)?;
+                        pipe_operators.push(PipeOperator::Select { exprs })
+                    }
+                    // EXTEND <expr> [[AS] alias], ...
+                    Keyword::EXTEND => {
+                        let exprs = self.parse_comma_separated(Parser::parse_select_item)?;
+                        pipe_operators.push(PipeOperator::Extend { exprs })
+                    }
+                    // SET <column> = <expression>, ...
+                    Keyword::SET => {
+                        let assignments = self.parse_comma_separated(Parser::parse_assignment)?;
+                        pipe_operators.push(PipeOperator::Set { assignments })
+                    }
+                    // DROP <column>, ...
+                    Keyword::DROP => {
+                        let columns = self.parse_identifiers()?;
+                        pipe_operators.push(PipeOperator::Drop { columns })
+                    }
+                    // AS <alias>
+                    Keyword::AS => {
+                        let alias = self.parse_identifier()?;
+                        pipe_operators.push(PipeOperator::Alias { alias })
+                    }
+                    // WHERE <condition>
+                    Keyword::WHERE => {
+                        let expr = self.parse_expr()?;
+                        pipe_operators.push(PipeOperator::Where { expr })
+                    }
+                    // LIMIT <n> [OFFSET <m>]
+                    Keyword::LIMIT => {
+                        let expr = self.parse_expr()?;
+                        let offset = if self.parse_keyword(Keyword::OFFSET) {
+                            Some(self.parse_expr()?)
+                        } else {
+                            None
+                        };
+                        pipe_operators.push(PipeOperator::Limit { expr, offset })
+                    }
+                    // AGGREGATE <agg_expr> [[AS] alias], ...
+                    //
+                    // and
+                    //
+                    // AGGREGATE [<agg_expr> [[AS] alias], ...]
+                    // GROUP BY <grouping_expr> [AS alias], ...
```

Comment:
Ah so I think these comments would be better placed on in the doc strings for each variant (left similar comment at that site)